students/C_Farris/session05/Mailroom_3.py

```diff
@@ -0,0 +1,211 @@
+
+import sys
+import os
+from textwrap import dedent
+
+OUT_PATH = "thank_you_letters"
+
+
+def prepare_to_write_to_disk():
+    """
+    Check OUT_PATH specified is a directory, if not, it will make it
+    This is to run once at the start of the application.
+    :Param: none
+    :Return: none. Creates specified folder in CWD if not already created.
+    """
+    if not os.path.isdir(OUT_PATH):
+        os.mkdir(OUT_PATH)
+
+
+def send_file_to_disk(person, myfile):
+    """
+    Accepts test file and writes to disk
+    :param (myfile):
+    :return: none. Will send file to directory set in outpath.
+    """
+    filename = person.replace(' ', '_') + '.txt'
+    filename = os.path.join(OUT_PATH, filename)
+    open(filename, 'w').write(myfile)
+
+
+def assemble_thank_you(getPerson='No_name_given',
+                       getDonation='No cash specified'):
+    """
+    assembles a thank you letter
+    :param(getPerson):
+    :param(getDonation):
+    :return: assembled text file with persons full name.txt
+    """
+    return dedent(
+        '''\tDear {},
+        Thank you for your generous donation of ${} to our cause
+        Because of donors like you, we are able to execute our mission
+        to support at risk youth in achieving academic success.
+        Your contribution also ensures we will have the funds to develop
+        new programs to reach those most vulnerable.
+        
+        Sincerely,
+        Team Umizoomi'''.format(getPerson, getDonation))
+
+
+def getDonation():
+    """
+    Asks user to specify donation amount and will error out if 
+    User specifies other than an int amount.
+    This could be modified in next version so entry can include 2 decimal places.
+    :Param: none
+    returns: donation retrieved from user
+    """
+    getDonation = ''
+    while not type(getDonation) is int:
+        try:
+            getDonation = int(input("Please enter donation amount:"
+                                    "==>"))
+        except ValueError:
+                print('Sorry, that isn\'t a valid integer. Please retry')
+    return getDonation
+
+
+def thankyou():
+    """
+    :param:
+    :returns:
+    """
+    getPerson = ''
+    donation = ''
+    while getPerson not in donor_db:
+        getPerson = input('Please type the donors first and last name or '
+                          'type list to get donor list ==>')
+        getPerson = getPerson.strip().lower().title()
+        if getPerson == 'List':
+            for key in donor_db:
+                print(key)
+        elif getPerson == 'x':
+            exitout()
+        else:
+            try:
+                donation = getDonation()
+                donor_db[getPerson].append(donation)
+            except KeyError:
+                print("This new person will be added to the dictionary.")
+                donation = getDonation()
+                donor_db[getPerson] = [donation]
+            finally:
+                printThankYou(getPerson, donation)
```

Comment:
Note that when a donor is not in the db, their donation amount is requested twice.  Can you identify why?